doc-migration/db2rst.py

```diff
@@ -0,0 +1,661 @@
+
+
+# lists
+
+def itemizedlist(el):
+    return _indent(el, 2, "* ", True)
+
+
+def orderedlist(el):
+    return _indent(el, 2, "1. ", True)
+
+
+def simplelist(el):
+    type = el.get("type")
+    if type == "inline":
+        return _join_children(el, ', ')
+    else:
+        return _concat(el)
+
+
+def member(el):
+    return _concat(el)
+
+# varlists
+
+
+def variablelist(el):
+    return _concat(el)
+
+
+def varlistentry(el):
+    s = ""
+    for i in el:
+        if i.tag == 'term':
+            s += _conv(i) + '\n\n'
+        else:
+            s += _indent(i, 3, None, True)
+    return s
+
+
+def listitem(el):
+    _supports_only(
+        el, ["para", "simpara", "{http://www.w3.org/2001/XInclude}include"])
+    return _block_separated_with_blank_line(el)
+
+# sections
+
+
+def example(el):
+    # FIXME: too hacky?
+    elements = [i for i in el]
+    first, rest = elements[0], elements[1:]
+
+    return _make_title(_concat(first), 3) + "\n\n" + "".join(_conv(i) for i in rest)
+
+
+def sect1(el):
+    return _block_separated_with_blank_line(el)
+
+
+def sect2(el):
+    return _block_separated_with_blank_line(el)
+
+
+def sect3(el):
+    return _block_separated_with_blank_line(el)
+
+
+def sect4(el):
+    return _block_separated_with_blank_line(el)
+
+
+def section(el):
+    return _block_separated_with_blank_line(el)
+
+
+def title(el):
+    return _make_title(_concat(el).strip(), _get_level(el) + 1)
+
+# bibliographic elements
+
+
+def author(el):
+    _has_only_text(el)
+    return "\n\n.. _author:\n\n**%s**" % el.text
+
+
+def date(el):
+    _has_only_text(el)
+    return "\n\n.. _date:\n\n%s" % el.text
+
+# references
+
+
+def citerefentry(el):
+    # Handles all external URl conversions from man/custom-html.xsl
+    project = el.get("project")
+    refentrytitle = el.xpath("refentrytitle")[0].text
+    manvolnum = el.xpath("manvolnum")[0].text
+    if project == 'man-pages' or manvolnum == 2 or manvolnum == 4:
+        return f"`{refentrytitle}({manvolnum}) <https://man7.org/linux/man-pages/man{manvolnum}/{refentrytitle}.{manvolnum}.html>`_"
+    if project == 'die-net':
+        return f"`{refentrytitle}({manvolnum}) <http://linux.die.net/man/ {manvolnum}/{refentrytitle}>`_"
+    if project == 'mankier':
+        return f"`{refentrytitle}({manvolnum}) <https://www.mankier.com/{manvolnum}/{refentrytitle}>`_"
+    if project == 'archlinux':
+        return f"`{refentrytitle}({manvolnum}) <https://man.archlinux.org/man/{refentrytitle}.{manvolnum}.en.html>`_"
+    if project == 'debian':
+        return f"`{refentrytitle}({manvolnum}) <https://manpages.debian.org/unstable/{refentrytitle}/{refentrytitle}.{manvolnum}.en.html>`_"
+    if project == 'freebsd':
+        return f"`{refentrytitle}({manvolnum}) <https://www.freebsd.org/cgi/man.cgi?{refentrytitle}({manvolnum})>`_"
+    if project == 'dbus':
+        return f"`{refentrytitle}({manvolnum}) <https://dbus.freedesktop.org/doc/{refentrytitle}.{manvolnum}.html>`_"
```

Comment:
I hope this can be avoided. There is a `:manpage:` role (https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-manpage). It only allows specifying one target URL, so this would need to be extended somehow…